codex-rs/apply-patch/src/lib.rs

```diff
@@ -780,9 +780,12 @@ fn compute_replacements(
     let mut line_index: usize = 0;
 
     for chunk in chunks {
-        // If a chunk has a `change_context`, we use seek_sequence to find it, then
-        // adjust our `line_index` to continue from there.
-        if let Some(ctx_line) = &chunk.change_context {
+        // If a chunk has one or more `change_context` lines, seek them in order
+        // to progressively narrow down the position of the chunk. This supports
+        // multiple @@ context headers such as:
+        // @@ class BaseClass:
+        // @@     def method():
+        for ctx_line in &chunk.change_context {
```

Comment:
I think we need a test where there are overlapping instances of the context?

For example, if the context were:

```
@@ A
@@ A
- old
```

and the file were:

```
A
A
A
old
```

I would want to be sure that the failed match for the first two `A`'s doesn't interfere with the match for the second and third `A`?